### PR TITLE
refactor: rename experience-builder to experiences-sdk-react

### PR DESCRIPTION
**Pull request by ryunsong-contentful**
## Purpose
Rename experience builder sdk packages based on https://contentful.atlassian.net/wiki/spaces/PROD/pages/4480499829/Experience+Builder+package+rename+proposal

This PR in particular converts `experience-builder` to `experiences-sdk-react`

## Approach
Create feature branch called experience-release based on the development branch.
Ignored the `CHANGELOG.md` and `package-lock.json` files as those should be auto regenerated

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -28,6 +28,8 @@ jobs:
           restore-keys: cypress-${{ runner.os }}-
       - name: Authenticate with private NPM package
         run: echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc
+        run: echo "@contentful:registry=https://npm.pkg.github.com" >> ~/.npmrc
+        run: echo "//npm.pkg.github.com/:_authToken=${{secrets.GH_TOKEN}}" >> ~/.npmrc
```
Comment (Chaoste) [inline .github/workflows/build.yaml:32]: This whole setup should not be necessary. This is a public repository that should only use public dependencies. Public packages would therefore be published to npmjs. Similar to our actual users, we should only fetch our node modules via npmjs to not mix up things in here.

Please be mindful to manually turn the package to "public" in the npmjs registry though :)

**Review by Chaoste — COMMENTED** [review]
Changes look good except for the github registry part

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ryunsong-contentful) [reply .github/workflows/build.yaml:32]: Sounds good - I'm going to make sure we publish it under the correct name as when you look it up it doesn't exist https://github.com/orgs/contentful/packages?tab=packages&q=experiences. So I'll have to rename it

Comment (Chaoste) [reply .github/workflows/build.yaml:32]: Note that we would still push this to github thanks to the `.npmrc` in this repo :) But this config here is only about installing modules which should still happen via npm

Comment (Chaoste) [reply .github/workflows/build.yaml:32]: Btw, so many packages with `experience` ^^ https://github.com/orgs/contentful/packages?q=experiences&tab=packages&q=experience

**Review by elylucas — APPROVED** [review]

**Pull request merged**